libs/application/ui-components/src/components/ApplicationList/ApplicationList.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import { MessageDescriptor } from '@formatjs/intl'
+import format from 'date-fns/format'
+
+import { ActionCard, Stack } from '@island.is/island-ui/core'
+import {
+  Application,
+  ApplicationStatus,
+  coreMessages,
+  getSlugFromType,
+  ActionCardTag,
+} from '@island.is/application/core'
+import { useLocale } from '@island.is/localization'
+import { dateFormat } from '@island.is/shared/constants'
+
+interface DefaultStateData {
+  tag: {
+    variant: ActionCardTag
+    label: MessageDescriptor
+  }
+  progress: {
+    variant: 'blue' | 'red' | 'rose' | 'mint'
+  }
+  cta: {
+    label: MessageDescriptor
+  }
+}
+
+const DefaultData: Record<ApplicationStatus, DefaultStateData> = {
+  [ApplicationStatus.REJECTED]: {
+    tag: {
+      variant: 'red',
+      label: coreMessages.tagsRejected,
+    },
+    progress: {
+      variant: 'red',
+    },
+    cta: {
+      label: coreMessages.cardButtonInProgress,
+    },
+  },
+  [ApplicationStatus.COMPLETED]: {
+    tag: {
+      variant: 'blueberry',
+      label: coreMessages.tagsDone,
+    },
+    progress: {
+      variant: 'mint',
+    },
+    cta: {
+      label: coreMessages.cardButtonComplete,
+    },
+  },
+  [ApplicationStatus.IN_PROGRESS]: {
+    tag: {
+      variant: 'blue',
+      label: coreMessages.tagsInProgress,
+    },
+    progress: {
+      variant: 'blue',
+    },
+    cta: {
+      label: coreMessages.cardButtonInProgress,
+    },
+  },
+}
+
+interface Props {
+  applications: Application[]
+  onClick: (id: string) => void
+}
+
+const ApplicationList = ({ applications, onClick }: Props) => {
+  const { lang: locale, formatMessage } = useLocale()
+  const formattedDate = locale === 'is' ? dateFormat.is : dateFormat.en
+
+  return (
+    <Stack space={2}>
+      {applications.map((application, index: number) => {
+        const actionCard = application.actionCard
+        const stateDefaultData =
+          DefaultData[application.status] ||
```

Comment:
I actually think the `||` operator is better here since we don't have any real scenario where the value of `DefaultData[application.status]` will be `0` or `''` 

It's definitely better in this example
`actionCard?.title ?? application.name`
Where the `actionCard?.title` could be set to an empty string intentionally and therefore we would not like to fallback to the `application.name`

I might be missing some case that is the reason for you going for `??` instead?

@jeremybarbet